ingest/scripts/compare_hashes.py

```diff
@@ -22,13 +23,177 @@ class Config:
     segmented: str
     nucleotide_sequences: list[str]
     debug_hashes: bool = False
+    slack_hook: str = ""
+
+
+InsdcAccession = str
+JointInsdcAccession = str
+LoculusAccession = str
+Status = str
+
+
+@dataclass
+class Revision:
+    loculus_accession: LoculusAccession
+    curated_fields: dict[str, str] | None = None  # map from field to value
+
+
+@dataclass
+class SequenceUpdateManager:
+    submit: list[JointInsdcAccession]
+    revise: dict[JointInsdcAccession, Revision]
+    noop: dict[JointInsdcAccession, LoculusAccession]
+    blocked: dict[Status, dict[JointInsdcAccession, LoculusAccession]]
+    revoke: dict[
+        JointInsdcAccession, dict[LoculusAccession, InsdcAccession]
+    ]  # Map of new grouping joint insdc accessions to map of previous state
+    # i.e. loculus accessions (to be revoked) and their corresponding old joint insdc accessions
+    sampled_out: list[JointInsdcAccession]
+    hashes: list[float]
+
+    def __init__(self, config: Config):
+        self.config = config
+        self.submit = []
+        self.revise = {}
+        self.noop = {}
+        self.blocked = defaultdict(dict)
+        self.revoke = {}
+        self.sampled_out = []
+        self.hashes = []
+
+
+def notify(config: Config, text: str):
+    """Send slack notification with blocked revision details"""
+    if config.slack_hook:
+        requests.post(config.slack_hook, data=json.dumps({"text": text}), timeout=10)
+    logger.warn(text)
+
+
+def get_curated_fields(curated_data, pre_curated_data, current_state):
+    curated_fields = {}
```

Comment:
we actually need not only the {field: curated_value} map but we should actually have {field: {to: value, from: value}}, then when checking for the re-ingested sequence where we want to change field x from y to z, we can check if the curation is from z to y - in which case we do not revise.